From 60a23d5042296ddf25f106490257c3662ae768ce Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 18 Oct 2018 17:42:42 +0300
Subject: [PATCH 0868/1200] ARM64: dts: armada-8040-db-G: Add dual PCIe x4
 configuration

Add DTS file for Armada-8040-DB setup G.
This configuration includes 1 PCIe x4 and one PCIe x1 port
on each of CP0 and CP1.

Change-Id: Iab43ff7f5810201a4122c907b643d8440015b300
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60383
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
---
 arch/arm/dts/Makefile                     |   1 +
 arch/arm/dts/armada-8040-db-G.dts         | 344 ++++++++++++++++++++++
 arch/arm/dts/armada-80x0-db.dtsi          |  15 +
 doc/mvebu/a7k-a8k/armada80x0-db-setup.txt | 292 +++++++++---------
 4 files changed, 511 insertions(+), 141 deletions(-)
 create mode 100644 arch/arm/dts/armada-8040-db-G.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 64b8c2ec46..c76a2ab21e 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -117,6 +117,7 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	armada-8040-db-D.dtb			\
 	armada-8040-db-E.dtb			\
 	armada-8040-db-F.dtb			\
+	armada-8040-db-G.dtb			\
 	armada-8040-mcbin.dtb			\
 	armada-8040-ocp.dtb			\
 	armada-xp-gp.dtb			\
diff --git a/arch/arm/dts/armada-8040-db-G.dts b/arch/arm/dts/armada-8040-db-G.dts
new file mode 100644
index 0000000000..964c71d064
--- /dev/null
+++ b/arch/arm/dts/armada-8040-db-G.dts
@@ -0,0 +1,344 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2018 Marvell International Ltd.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 8040 Development board platform
+ */
+
+#include "armada-80x0-db.dtsi"
+
+/ {
+	model = "Marvell Armada 8040 Development Board DB-88F8040 setup(G)";
+	compatible = "marvell,armada8040-db", "marvell,armada8040",
+		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		i2c0 = &cp0_i2c0;
+		spi0 = &cp1_spi1;
+		fuse1 = &ap_ld_efuse0;
+		fuse2 = &ap_ld_efuse1;
+		fuse3 = &cp0_ld_efuse0;
+		fuse4 = &cp0_ld_efuse1;
+		fuse5 = &cp1_ld_efuse0;
+		fuse6 = &cp1_ld_efuse1;
+	};
+
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+};
+
+/* Accessible over the mini-USB CON9 connector on the main board */
+&uart0 {
+	status = "okay";
+};
+
+&ap_hd_efuse0 {
+	status = "okay";
+};
+
+&ap_ld_efuse0 {
+	status = "okay";
+};
+
+&ap_ld_efuse1 {
+	status = "okay";
+};
+
+&ap_pinctl {
+	/* MPP Bus:
+	 * SDIO  [0-10]
+	 * UART0 [11,19]
+	 */
+		  /* 0 1 2 3 4 5 6 7 8 9 */
+	pin-func = < 1 1 1 1 1 1 1 1 1 1
+		     1 3 0 0 0 0 0 0 0 3 >;
+};
+
+&ap_sdhci0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ap_emmc_pins>;
+	bus-width = <8>;
+	status = "okay";
+};
+
+&cp0_pinctl {
+	/* MPP Bus:
+	 *	[0-31]	= 0xff: Keep default CP0_shared_pins
+	 *	[11]	CLKOUT_MPP_11 (out)
+	 *	[23]	LINK_RD_IN_CP2CP (in)
+	 *	[25]	CLKOUT_MPP_25 (out)
+	 *	[29]	AVS_FB_IN_CP2CP (in)
+	 *	[32,34]	GE_MDIO/MDC
+	 *	[33]	GPIO: GE_INT#/push button/Wake
+	 *	[35]	MSS_GPIO[3]: MSS_PWDN
+	 *	[36]	MSS_GPIO[5]: MSS_VTT_EN
+	 *	[37-38]	I2C0
+	 *	[39]	PTP_CLK
+	 *	[40-41]	SATA[0/1]_PRESENT_ACTIVEn
+	 *	[42-43]	XG_MDC/XG_MDIO (XSMI)
+	 *	[44-55]	RGMII1
+	 *	[56-62]	SD
+	 */
+	/*   0    1    2    3    4    5    6    7    8    9 */
+	pin-func = < 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+		     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+		     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+		     0xff 0xff 0x7  0x0  0x7  0xa  0xa  0x2  0x2  0x5
+		     0x9  0x9  0x8  0x8  0x1  0x1  0x1  0x1  0x1  0x1
+		     0x1  0x1  0x1  0x1  0x1  0x1  0xe  0xe  0xe  0xe
+		     0xe  0xe  0xe>;
+};
+
+&cp0_ld_efuse0 {
+	status = "okay";
+};
+
+&cp0_ld_efuse1 {
+	status = "okay";
+};
+
+&cp0_comphy {
+	/* Serdes Configuration:
+	 *	Lane 0-3: PCIe0 (x4)
+	 *	Lane 4: USB3_HOST1
+	 *	Lane 5: PCIe2 (x1)
+	 */
+	phy0 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+	};
+
+	phy1 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+	};
+
+	phy2 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+	};
+
+	phy3 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+	};
+
+	phy4 {
+		phy-type = <COMPHY_TYPE_USB3_HOST1>;
+	};
+
+	phy5 {
+		phy-type = <COMPHY_TYPE_PEX2>;
+	};
+};
+
+/* CON6 on CP0 expansion */
+&cp0_pcie0 {
+	status = "okay";
+};
+
+&cp0_pcie1 {
+	status = "disabled";
+};
+
+/* CON5 on CP0 expansion */
+&cp0_pcie2 {
+	status = "okay";
+};
+
+&cp0_i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_i2c0_pins>;
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+/* CON9 on CP0 expansion */
+&cp0_usb3_0 {
+	status = "okay";
+};
+
+/* CON10 on CP0 expansion */
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&cp0_utmi0 {
+	status = "okay";
+};
+
+&cp0_utmi1 {
+	status = "okay";
+};
+
+&cp0_sdhci0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_sdhci_pins>;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&cp0_mdio {
+	status = "okay";
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+};
+
+&cp1_pinctl {
+	/* MPP Bus:
+	 *	[0-11]	RGMII0
+	 *	[13-16]	SPI1
+	 *	[27,31]	GE_MDIO/MDC
+	 *	[28]	SATA1_PRESENT_ACTIVEn
+	 *	[29-30]	UART0
+	 *	[32-62]	= 0xff: Keep default CP1_shared_pins
+	 */
+	/*   0    1    2    3    4    5    6    7    8    9 */
+	pin-func = < 0x3  0x3  0x3  0x3  0x3  0x3  0x3  0x3  0x3  0x3
+		     0x3  0x3  0x3  0x3  0x3  0x3  0x3  0xff 0xff 0xff
+		     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0x8  0x9  0xa
+		     0xA  0x8  0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+		     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+		     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+		     0xff 0xff 0xff>;
+};
+
+&cp1_ld_efuse0 {
+	status = "okay";
+};
+
+&cp1_ld_efuse1 {
+	status = "okay";
+};
+
+&cp1_comphy {
+	/* Serdes Configuration:
+	 *	Lane 0-3: PCIe0 (x4)
+	 *	Lane 4: USB3_HOST1
+	 *	Lane 5: PCIe2 (x1)
+	 */
+	phy0 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+	};
+
+	phy1 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+	};
+
+	phy2 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+	};
+
+	phy3 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+	};
+
+	phy4 {
+		phy-type = <COMPHY_TYPE_USB3_HOST1>;
+	};
+
+	phy5 {
+		phy-type = <COMPHY_TYPE_PEX2>;
+	};
+};
+
+/* CON6 on CP1 expansion */
+&cp1_pcie0 {
+	status = "okay";
+};
+
+&cp1_pcie1 {
+	status = "disabled";
+};
+
+/* CON5 on CP1 expansion */
+&cp1_pcie2 {
+	status = "okay";
+};
+
+&cp1_spi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp1_spi1_pins>;
+	reg = <0x700680 0x50>,		/* control */
+	      <0x2000000 0x1000000>,	/* CS0 */
+	      <0 0xffffffff>;		/* CS1 */
+	status = "okay";
+
+	spi-flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor", "spi-flash";
+		reg = <0>;
+		spi-max-frequency = <108000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "U-Boot";
+				reg = <0 0x200000>;
+			};
+			partition@400000 {
+				label = "Filesystem";
+				reg = <0x200000 0xce0000>;
+			};
+		};
+	};
+};
+
+/* CON9 on CP1 expansion */
+&cp1_usb3_0 {
+	status = "okay";
+};
+
+/* CON10 on CP0 expansion */
+&cp1_usb3_1 {
+	status = "okay";
+};
+
+&cp1_utmi0 {
+	status = "okay";
+};
+
+&cp1_utmi1 {
+	status = "okay";
+};
+
+&cp1_mdio {
+	status = "okay";
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+};
+
+&cp1_ethernet {
+	status = "okay";
+};
+
+&cp1_eth0 {
+	status = "disabled";
+	phy-mode = "sfi"; /* lane-2 */
+};
+
+&cp1_eth1 {
+	status = "okay";
+	phy = <&phy0>;
+	phy-mode = "rgmii-id";
+};
diff --git a/arch/arm/dts/armada-80x0-db.dtsi b/arch/arm/dts/armada-80x0-db.dtsi
index b48596a4e0..fe074e07bf 100644
--- a/arch/arm/dts/armada-80x0-db.dtsi
+++ b/arch/arm/dts/armada-80x0-db.dtsi
@@ -78,6 +78,15 @@
 				gpio = <&expander1 0 GPIO_ACTIVE_HIGH>;
 			};
 
+			cp0_reg_usb3_vbus3: cp0_usb3_vbus@3 {
+				compatible = "regulator-fixed";
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5000000>;
+				startup-delay-us = <100000>;
+				regulator-force-boot-off;
+				gpio = <&expander1 1 GPIO_ACTIVE_HIGH>;
+			};
+
 			cp0_reg_usb3_current_limiter2: cp0_usb3_current_limiter@2 {
 				compatible = "regulator-fixed";
 				regulator-min-microamp = <900000>;
@@ -122,6 +131,12 @@
 				current-limiter = <&cp0_reg_usb3_current_limiter2>;
 				vbus-disable-delay = <500>;
 			};
+
+			cp1_usb3_1: usb3@510000 {
+				vbus-supply = <&cp0_reg_usb3_vbus3>;
+				vbus-disable-delay = <500>;
+			};
+
 		};
 	};
 #undef CP110_NUM
diff --git a/doc/mvebu/a7k-a8k/armada80x0-db-setup.txt b/doc/mvebu/a7k-a8k/armada80x0-db-setup.txt
index 791188fba2..03101de557 100644
--- a/doc/mvebu/a7k-a8k/armada80x0-db-setup.txt
+++ b/doc/mvebu/a7k-a8k/armada80x0-db-setup.txt
@@ -28,63 +28,63 @@ Booting Linux requires using the same Device Tree configuration as used in U-Boo
 Board Setup Options
 -------------------
 
-+---------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|		| 1: A			| 2: B			| 3: C			| 4: D			| 5: E			| 6: F			|
-+===============+=======================+=======================+=======================+=======================+=======================+=======================+
-|Device tree	|armada-8040-db.dtb	|armada-8040-db-B.dtb	|armada-8040-db-C.dtb	|armada-8040-db-D.dtb	|armada-8040-db-E.dtb	|armada-8040-db-F.dtb	|
-+-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-| CP0 | SW1:1	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:2	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:3	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:4	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:5	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:6	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:7	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-+-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-| CP1 | SW1:1	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:2	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:3	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:4	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:5	| ON			| ON			| ON			| ON			| ON			| ON			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:6	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:7	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
-+-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
++---------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|		| 1: A			| 2: B			| 3: C			| 4: D			| 5: E			| 6: F			| 7: G			|
++===============+=======================+=======================+=======================+=======================+=======================+=======================+=======================+
+|Device tree	|armada-8040-db.dtb	|armada-8040-db-B.dtb	|armada-8040-db-C.dtb	|armada-8040-db-D.dtb	|armada-8040-db-E.dtb	|armada-8040-db-F.dtb	|armada-8040-db-G.dtb	|
++-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| CP0 | SW1:1	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:2	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:3	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:4	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:5	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:6	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:7	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
++-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| CP1 | SW1:1	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:2	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:3	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:4	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:5	| ON			| ON			| ON			| ON			| ON			| ON			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:6	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:7	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
++-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
 
 DPR Setting
 ~~~~~~~~~~~
 
-+-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+
-|						| 1: A		| 2: B		| 3: C			| 4: D			| 5: E				| 6: F		|
-+===============================================+===============+===============+=======================+=======================+===============================+===============+
-| Jumper-module (CON44): SLM1564		| DPR all = 1-2	| DPR all = 1-2	| DPR 96-101 = 2-3,	| DPR 96-101 = 2-3,	| DPR 109-112 = 2-3,		| DPR all = 1-2	|
-|						|		|		| DPR 103-108 = 2-3,	| DPR 103-108 = 2-3,	| others = 1-2			|		|
-|						|		|		| DPR 102 = 1-2		| DPR 102 = 1-2		|				|		|
-+-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+
-| Serdes CP0 site module (CON42): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 2-3	| DPR 2-5 = 1-2		|DPR 2-5 = 1-2		| DPR 2-5 = 1-2			| DPR 2-5 = 1-2	|
-+-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+
-| Serdes CP1 site module (CON41): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2		|DPR 2-5 = 1-2		| DPR 2-5 = 1-2			| DPR 2-5 = 1-2	|
-+-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+
-| DB8040					|		|		| DPR 84 = 2-3		| DPR 76 = 1-2,		| DPR 81 = 2-3,			|		|
-|						|		|		| TDM module (CON51):	| DPR 87-88 = 2-3	| DPR 23-26,119,121 =1-2	|		|
-|						|		|		| SLM1448		| Audio module (CON51):	|				|		|
-|						|		|		|			| SLM1449		|				|		|
-+-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+
++-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+---------------+
+|						| 1: A		| 2: B		| 3: C			| 4: D			| 5: E				| 6: F		| 7: G		|
++===============================================+===============+===============+=======================+=======================+===============================+===============+===============+
+| Jumper-module (CON44): SLM1564		| DPR all = 1-2	| DPR all = 1-2	| DPR 96-101 = 2-3,	| DPR 96-101 = 2-3,	| DPR 109-112 = 2-3,		| DPR all = 1-2	|DPR all = 1-2	|
+|						|		|		| DPR 103-108 = 2-3,	| DPR 103-108 = 2-3,	| others = 1-2			|		|		|
+|						|		|		| DPR 102 = 1-2		| DPR 102 = 1-2		|				|		|		|
++-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+---------------+
+| Serdes CP0 site module (CON42): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 2-3	| DPR 2-5 = 1-2		|DPR 2-5 = 1-2		| DPR 2-5 = 1-2			| DPR 2-5 = 1-2	| DPR 2-5 = 2-3	|
++-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+---------------+
+| Serdes CP1 site module (CON41): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2		|DPR 2-5 = 1-2		| DPR 2-5 = 1-2			| DPR 2-5 = 1-2	| DPR 2-5 = 1-2	|
++-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+---------------+
+| DB8040					|		|		| DPR 84 = 2-3		| DPR 76 = 1-2,		| DPR 81 = 2-3,			|		|		|
+|						|		|		| TDM module (CON51):	| DPR 87-88 = 2-3	| DPR 23-26,119,121 =1-2	|		|		|
+|						|		|		| SLM1448		| Audio module (CON51):	|				|		|		|
+|						|		|		|			| SLM1449		|				|		|		|
++-----------------------------------------------+---------------+---------------+-----------------------+-----------------------+-------------------------------+---------------+---------------+
 
 The tables below summarizes the interface configuration of each setup
 
@@ -92,40 +92,40 @@ The tables below summarizes the interface configuration of each setup
 SerDes Configurations
 ~~~~~~~~~~~~~~~~~~~~~
 
-+---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+
-| CP0 Lane	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F			|
-+===============+===============+===============+===============+===============+===============+=======================+
-| 0		| PCIe0 (x1)	| PCIe0 (x4)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)		|
-+---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+
-| 1		| SATA0		| PCIe0 (x4)	| SATA0		| SATA0		| SATA0		| SATA0			|
-+---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+
-| 2		| SFI0 (10G)	| PCIe0 (x4)	| SFI0 (10G)	| SFI0 (10G)	| SFI0 (10G)	| AP (802.3 auto-neg)	|
-+---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+
-| 3		| SATA1		| PCIe0 (x4)	| SATA1		| SATA1		| SATA1		| SATA1			|
-+---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+
-| 4		| USB3_HOST1	| USB3_HOST1	| USB3_HOST1	| USB3_HOST1	| USB3_HOST1	| USB3_HOST1		|
-+---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+
-| 5		| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)		|
-+---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+
++---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+---------------+
+| CP0 Lane	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F			| 7: G		|
++===============+===============+===============+===============+===============+===============+=======================+===============+
+| 0		| PCIe0 (x1)	| PCIe0 (x4)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)		| PCIe0 (x4)	|
++---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+---------------+
+| 1		| SATA0		| PCIe0 (x4)	| SATA0		| SATA0		| SATA0		| SATA0			| PCIe0 (x4)	|
++---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+---------------+
+| 2		| SFI0 (10G)	| PCIe0 (x4)	| SFI0 (10G)	| SFI0 (10G)	| SFI0 (10G)	| AP (802.3 auto-neg)	| PCIe0 (x4)	|
++---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+---------------+
+| 3		| SATA1		| PCIe0 (x4)	| SATA1		| SATA1		| SATA1		| SATA1			| PCIe0 (x4)	|
++---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+---------------+
+| 4		| USB3_HOST1	| USB3_HOST1	| USB3_HOST1	| USB3_HOST1	| USB3_HOST1	| USB3_HOST1		| USB3_HOST1	|
++---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+---------------+
+| 5		| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)		| PCIe2 (x1)	|
++---------------+---------------+---------------+---------------+---------------+---------------+-----------------------+---------------+
 
 - USB2_H0 (UTMI only)
 - USB2_H1 (UTMI & USB3 SerDes)
 
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| CP1 Lane	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		|
-+===============+===============+===============+===============+===============+===============+===============+
-| 0		| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| 1		| SATA0		| USB3_HOST0	| SATA0		| SATA0		| SATA0		| SATA0		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| 2		| SFI0 (10G)	| SFI0 (10G)	| SFI0 (10G)	| SGMII0	| SGMII0 (2.5G)	| SFI0 (10G)	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| 3		| SATA1		| SATA1		| SATA1		| SATA1		| SATA1		| SATA1		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| 4		| PCIe1 (x1)	| PCIe1 (x1)	| PCIe1 (x1)	| PCIe1 (x1)	| PCIe1 (x1)	| PCIe1 (x1)	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| 5		| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| CP1 Lane	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		| 7: G		|
++===============+===============+===============+===============+===============+===============+===============+===============+
+| 0		| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x1)	| PCIe0 (x4)	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 1		| SATA0		| USB3_HOST0	| SATA0		| SATA0		| SATA0		| SATA0		| PCIe0 (x4)	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 2		| SFI0 (10G)	| SFI0 (10G)	| SFI0 (10G)	| SGMII0	| SGMII0 (2.5G)	| SFI0 (10G)	| PCIe0 (x4)	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 3		| SATA1		| SATA1		| SATA1		| SATA1		| SATA1		| SATA1		| PCIe0 (x4)	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 4		| PCIe1 (x1)	| PCIe1 (x1)	| PCIe1 (x1)	| PCIe1 (x1)	| PCIe1 (x1)	| PCIe1 (x1)	| USB3_HOST1	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 5		| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	| PCIe2 (x1)	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
 
 - USB2_H0 (UTMI only)
 
@@ -133,70 +133,70 @@ SerDes Configurations
 Multi-purpose pin configurations
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| AP806 pin	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		|
-+===============+===============+===============+===============+===============+===============+===============+
-|    AP-SDIO	| [0-10]	| [0-10]	| [0-10]	| [0-10]	| N/C		| [0-10]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|    I2C0	| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|    AP-SPI0	| N/C		| N/C		| N/C		| N/C		| [0,3]		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|    AP-UART0	| [11,19]	| [11,19]	| [11,19]	| [11,19]	| [11,19]	| [11,19]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| CP0 pin	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		|
-+===============+===============+===============+===============+===============+===============+===============+
-|   CP0-SMI	| [32,34]	| [32,34]	| [32,34]	| [32,34]	| N/C		| [32,34]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-XSMI	| [42-43]	| [42-43]	| [42-43]	| [42-43]	| [42-43]	| [42-43]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-TWSI	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-SATA1	| [40]		| [40]		| [40]		| [40]		| [40]		| [40]		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-SATA0	| [41]		| [41]		| [41]		| [41]		| [41]		| [41]		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-RGMII1	| [44-55]	| [44-55]	| [44-55]	| [44-55]	| N/C		| [44-55]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-SD	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-PTP_CLK	| [39]		| [39]		| [39]		| [39]		| [39]		| [39]		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-TWSI1	| N/C		| N/C		| N/C		| N/C		| [35-36]	| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-UART1	| N/C		| N/C		| N/C		| N/C		| [47,49]	| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP0-UART2	| N/C		| N/C		| N/C		| N/C		| [50-51]	| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| AP806 pin	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		| 7: G		|
++===============+===============+===============+===============+===============+===============+===============+===============+
+|    AP-SDIO	| [0-10]	| [0-10]	| [0-10]	| [0-10]	| N/C		| [0-10]	| [0-10]
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|    I2C0	| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|    AP-SPI0	| N/C		| N/C		| N/C		| N/C		| [0,3]		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|    AP-UART0	| [11,19]	| [11,19]	| [11,19]	| [11,19]	| [11,19]	| [11,19]	| [11,19]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| CP0 pin	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		| 7: G		|
++===============+===============+===============+===============+===============+===============+===============+===============+
+|   CP0-SMI	| [32,34]	| [32,34]	| [32,34]	| [32,34]	| N/C		| [32,34]	| [32,34]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-XSMI	| [42-43]	| [42-43]	| [42-43]	| [42-43]	| [42-43]	| [42-43]	| [42-43]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-TWSI	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-SATA1	| [40]		| [40]		| [40]		| [40]		| [40]		| [40]		| [40]		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-SATA0	| [41]		| [41]		| [41]		| [41]		| [41]		| [41]		| [41]		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-RGMII1	| [44-55]	| [44-55]	| [44-55]	| [44-55]	| N/C		| [44-55]	| [44-55]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-SD	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-PTP_CLK	| [39]		| [39]		| [39]		| [39]		| [39]		| [39]		| [39]		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-TWSI1	| N/C		| N/C		| N/C		| N/C		| [35-36]	| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-UART1	| N/C		| N/C		| N/C		| N/C		| [47,49]	| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP0-UART2	| N/C		| N/C		| N/C		| N/C		| [50-51]	| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
 
 GPIOS:
 	- 33: GPIO: GE_INT#/push button/Wake
 	- 35: MSS_GPIO[3]: MSS_PWDN (only in configuration 1,3,4,5)
 	- 36: MSS_GPIO[5]: MSS_VTT_EN (only in configuration 1,3,4,5)
 
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-| CP1 pin	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		|
-+===============+===============+===============+===============+===============+===============+===============+
-|   CP1-RGMII0	| [0-11]	| [0-11]	| N/C		| N/C		| [0-11]	| [0-11]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP1-SMI	| [27,31]	| [27,31]	| N/C		| N/C		| [27,31]	| [27,31]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP1-SPI1	| [13-16]	| [13-16]	| [13-16]	| N/C		| [13-16]	| [13-16]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP1-UART0	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP1-SATA1	| [28]		| [28]		| [28]		| [28]		| [28]		| [28]		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP1-SATA0	| N/C		| N/C		| [27]		| N/C		| N/C		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP1-AU	| N/C		| N/C		| N/C		| [0-5]		| N/C		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP1-NAND	| N/C		| N/C		| N/C		| [13,15-27]	| N/C		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
-|   CP1-TDM	| N/C		| N/C		| [0-11]	| N/C		| N/C		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| CP1 pin	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		| 7: G		|
++===============+===============+===============+===============+===============+===============+===============+===============+
+|   CP1-RGMII0	| [0-11]	| [0-11]	| N/C		| N/C		| [0-11]	| [0-11]	| [0-11]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP1-SMI	| [27,31]	| [27,31]	| N/C		| N/C		| [27,31]	| [27,31]	| [27,31]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP1-SPI1	| [13-16]	| [13-16]	| [13-16]	| N/C		| [13-16]	| [13-16]	| [13-16]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP1-UART0	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP1-SATA1	| [28]		| [28]		| [28]		| [28]		| [28]		| [28]		| [28]		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP1-SATA0	| N/C		| N/C		| [27]		| N/C		| N/C		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP1-AU	| N/C		| N/C		| N/C		| [0-5]		| N/C		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP1-NAND	| N/C		| N/C		| N/C		| [13,15-27]	| N/C		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|   CP1-TDM	| N/C		| N/C		| [0-11]	| N/C		| N/C		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+
 
 Network configuration
 ~~~~~~~~~~~~~~~~~~~~~
@@ -277,5 +277,15 @@ Setup 6 (F):
 | eth3		| 1	| 1		| 2		| RGMII0					|
 +---------------+-------+---------------+---------------+-----------------------------------------------+
 
+Setup 7 (G):
+
++---------------+-------+---------------+---------------+-------------------------------+
+| Interface	| CP#	| PPv2 Port	| GMAC Port	| Board Interface		|
++===============+=======+===============+===============+===============================+
+| eth0		| 0	| 2		| 3		| RGMII1			|
++---------------+-------+---------------+---------------+-------------------------------+
+| eth1		| 1	| 1		| 2		| RGMII0			|
++---------------+-------+---------------+---------------+-------------------------------+
+
 Notes:
 	- eth0/1/2/3 may in fact be higher numbers, if prior eth_x interfaces already exist.
-- 
2.22.0

